src/Flagged/Modal.jsx

```diff
@@ -0,0 +1,97 @@
+let { onReport, open, reportedAccountId, onOpenChange } = props;
+
+const profileUrl = `#/${REPL_ACCOUNT}/widget/ProfilePage?accountId=${reportedAccountId}`;
+
+// copied from Moderate.jsx so maybe it's wrong
+const getFlaggedAccountsList = Social.get(
+  `${context.accountId}/moderate/users`,
+  "optimistic"
+);
+
+const reportAccount = () => {
+  const contentModerationItem = {
+    type: "social",
+    path: profileUrl,
+    reportedBy: context.accountId,
+  };
+
+  Social.set(
+    {
+      index: {
+        flag: JSON.stringify({
+          key: "main",
+          value: contentModerationItem,
+        }),
+      },
+    },
+    {
+      onCommit: () => {
+        onReport && onReport();
+      },
+      onCancel: () => onOpenChange(false),
+    }
+  );
+};
+
+const reportAccountWithPosts = () => {
+  const flaggedAccountsList = getFlaggedAccountsList
+    ? JSON.parse(getFlaggedAccountsList)
+    : [];
+  const flaggedAccountsListSet = new Set(flaggedAccountsList);
+  const filterFlaggedAccounts = [
+    ...flaggedAccountsListSet.add(reportedAccountId),
+  ];
+
+  Social.set(
+    {
+      index: {
+        moderate: {
+          hide: {
```

Comment:
Can you switch to the new structure described [here](https://github.com/near/near-discovery/discussions/386)?

```suggestion
         [filterFlaggedAccounts]: ""
```

Reply:
And it will mean that we want ti hide all the posts without reporting, right @gabehamilton ?

Reply:
Yes, that's correct.